storage/txcache/txListForSender.go

```diff
@@ -184,20 +196,81 @@ func (listForSender *txListForSender) findListElementWithTx(txToFind *WrappedTra
 	txToFindHash := txToFind.TxHash
 	txToFindNonce := txToFind.Tx.GetNonce()
 
+	iterations := 0
+	defer listForSender.monitorOnFoundTransaction(&iterations, txToFind)
+
 	for element := listForSender.items.Front(); element != nil; element = element.Next() {
-		value := element.Value.(*WrappedTransaction)
+		iterations++
 
-		if bytes.Equal(value.TxHash, txToFindHash) {
-			return element
+		value := element.Value.(*WrappedTransaction)
+		nonce := value.Tx.GetNonce()
+
+		// Optimization: first check nonce equality.
+		if nonce == txToFindNonce {
+			if bytes.Equal(value.TxHash, txToFindHash) {
+				return element
+			}
+		} else if nonce > txToFindNonce {
+			// Optimization: stop search at this point, since the list is sorted by nonce.
+			break
 		}
+	}
+
+	return nil
+}
 
-		// Optimization: stop search at this point, since the list is sorted by nonce
-		if value.Tx.GetNonce() > txToFindNonce {
+// RemoveSortedTransactions removes the provided list of transactions, assuming the list is given sorted by nonce
+// QUESTION FOR REVIEW: should I refactor to have a loop-in-loop when traversing the two lists instead?
+func (listForSender *txListForSender) RemoveSortedTransactions(txsToRemove []*WrappedTransaction) int {
+	listForSender.mutex.Lock()
+	defer listForSender.mutex.Unlock()
+
+	numRemoved := 0
+	pointerInTxsToRemove := 0
+	pointerInTxs := listForSender.items.Front()
+
+	for {
+		// End of any of the two lists, therefore end of loop
+		if pointerInTxs == nil || pointerInTxsToRemove == len(txsToRemove) {
```

Comment:
can you move the negated condition to the for? then no need for break condition inside loop 
```go
for pointerInTxs != nil && pointerInTxsToRemove != len(txsToRemove) {
    ...
}
```

Reply:
Fixed, thanks :)

> break and continue "are just the old goto in sheep's clothing" / _Wise man, 20th century_